src/modules/layoutUiRules/__tests__/rules.spec.js

```diff
@@ -1,40 +1,182 @@
-                )
+            onlyRulesWithProp(ruleProp).every(
+                rule =>
+                    Object.keys(rule[ruleProp]).length &&
+                    Object.values(rule[ruleProp]).length
             )
         ).toBe(true)
     })
 
-    it('should return false (mocked rules)', () => {
-        const mockRules = [{}]
+const testPropIsArray = ruleProp =>
+    it('is an array', () => {
+        expect(
+            onlyRulesWithProp(ruleProp).every(rule =>
+                Array.isArray(rule[ruleProp])
+            )
+        ).toBe(true)
+    })
 
+const testKeysAreValidAxisIds = ruleProp =>
+    it('keys should be valid axis ids', () => {
         expect(
-            mockRules.every(rule =>
-                testResourceRequiredProps.every(prop =>
-                    prop.isValid(rule[prop.name])
-                )
+            onlyRulesWithProp(ruleProp).every(rule =>
+                allArrayItemsAreValidAxisIds(Object.keys(rule[ruleProp]))
             )
-        ).toBe(false)
+        ).toBe(true)
     })
 
-    it('should return false (mocked required props)', () => {
-        const mockRequiredProps = [
-            {
-                name: 'notRealRequiredProp',
-                isValid: prop => prop !== undefined,
-            },
-        ]
+const testNoValuesZero = ruleProp =>
+    it('values should not be 0', () => {
+        expect(
+            onlyRulesWithProp(ruleProp).every(rule =>
+                Object.values(rule[ruleProp]).every(value => value !== 0)
+            )
+        ).toBe(true)
+    })
+
+const testNoValuesNegative = ruleProp =>
+    it('values should not be negative', () => {
+        expect(
+            onlyRulesWithProp(ruleProp).every(rule =>
+                Object.values(rule[ruleProp]).every(value => value >= 0)
+            )
+        ).toBe(true)
+    })
+
+// Test implementations
+describe('verify test resource', () => {
+    it('testResourceRules is array', () => {
+        expect(Array.isArray(testResourceRules)).toBe(true)
+    })
+
+    it('testResourceRequiredProps is array', () => {
+        expect(Array.isArray(testResourceRequiredProps)).toBe(true)
+    })
+})
+
+describe('verify all rules', () => {
+    it('implements all required props', () => {
+        expect(
+            testResourceRules.every(rule =>
+                testResourceRequiredProps.every(
+                    val => Object.keys(rule).indexOf(val) !== -1
+                )
+            )
+        ).toBe(true)
+    })
 
+    it('only implements allowed rule props', () => {
         expect(
             testResourceRules.every(rule =>
-                mockRequiredProps.every(prop => prop.isValid(rule[prop.name]))
+                allArrayItemsAreValid(
+                    Object.keys(rule),
+                    Object.values(testResourceAllRuleProps)
+                )
             )
-        ).toBe(false)
+        ).toBe(true)
+    })
+})
+
+describe("verify each rule's ", () => {
```

Comment:
?

Reply:
This was added to improve the readability of the test output and enables the following:
![Screenshot 2019-12-05 at 10 25 19](https://user-images.githubusercontent.com/12590483/70222069-b54d1280-1749-11ea-84e7-6459366ab0fa.png)